templates/admin/settings/fields/checkbox-post-types.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * Settings post types checkbox template.
+ *
+ * @package AMP
+ */
+
+// Check referrer.
+if ( ! $this instanceof AMP_Settings_Post_Types ) {
+	return;
+}
+?>
+<fieldset>
+	<?php foreach ( $this->get_supported_post_types() as $post_type ) : ?>
+		<label>
+			<input type="checkbox" value="1" name="<?php echo esc_attr( $this->get_setting_name( $post_type->name ) ); ?>"<?php checked( true, (bool) $this->get_settings_value( $post_type->name ) || post_type_supports( $post_type->name, AMP_QUERY_VAR ) ); ?><?php disabled( true, post_type_supports( $post_type->name, AMP_QUERY_VAR ) ); ?>> <?php echo esc_html( $post_type->label ); ?>
```

Comment:
My comment about `is_enabled_via_setting` above will impact this. In particular, the `disabeld()` call would need to be changed to:

```php
disabled( true, post_type_supports( $post_type->name, AMP_QUERY_VAR ) && ! $this->get_settings_value( $post_type->name ) )
```